release-blocking-jobs.md

```diff
@@ -2,56 +2,122 @@
+
+The thresholds given are aspirational, and we recognize that not every job may
+be able to strictly adhere to all of these criteria all of the time. Thus we
+consider human review a key component in both the addition and removal of a
+job's release-blocking status.
+
+We intend to drive the criteria to stricter thresholds over time, e.g.,
+- finishing a run in 60min or less
+- running at least every 2 hours
+- passing at least 90% of all runs1
+
+## Release Informing Criteria
+
+Jobs that are considered useful or necessary to inform whether a commit on
+master is ready for release, but that clearly do not meet release-blocking 
+criteria, may be placed on the sig-release-master-informing dashboard. These
+are often jobs that are some combination of slow, flaky, infrequent, or
+difficult to maintain.
+
+These jobs may still block the release, but because they require a lot of
+manual, human interpretation, we choose to move them to a separate dashboard.
+
+<!-- TOOD(spiffxp): implement release-master-informing, populate with:
+- gce-master-scale-correctness
+- gce-master-scale-performance
+- gce-cos-master-serial
+- gke-cos-master-serial
+- packages-pushed-master
+- conformance jobs that run less frequently than every 3 hours
+-->
+
+## The release-blocking dashboards
 
 <!-- TODO(krzyzacy): use master-blocking dashboard, once we sort out jobs like scalability -->
 
 We use [1.12 release-blocking dashboard](http://k8s-testgrid.appspot.com/sig-release-1.12-blocking)
-as a source of truth:
+as a source of truth. The tab names are autogenerated based on the job names,
+and have the branch/version in their name. For example, for release x.y:
 
 * Make & Unit & Integration
-  - build
-  - verify
-  - integration
-  - bazel-build
-  - bazel-test
+  - build-x.y
+  - verify-x.y
+  - integration-x.y
+  - bazel-build-x.y
+  - bazel-test-x.y
 * GCE
-  - gce-cos-default
-  - gce-cos-serial
-  - gce-cos-slow
-  - gce-cos-ingress
-  - gce-cos-reboot
-  - gce-cos-alphafeatures
-  - gce-cos-conformance
-  - gce-gpu-device-plugin
-  - gce-scalability
+  - gce-cos-default-x.y
+  - gce-cos-serial-x.y
+  - gce-cos-slow-x.y
+  - gce-cos-ingress-x.y
+  - gce-cos-reboot-x.y
+  - gce-cos-alphafeatures-x.y
+  - gce-cos-conformance-x.y
+  - gce-gpu-device-plugin-x.y
+  - gci-gce-scalability-x.y
 * GKE
-  - gke-cos-default
-  - gke-cos-serial
-  - gke-cos-slow
-  - gke-cos-ingress
-  - gke-cos-reboot
-  - gke-gpu-device-plugin
-  - gke-gpu-device-plugin-p100
+  - gke-cos-default-x.y
+  - gke-cos-serial-x.y
+  - gke-cos-slow-x.y
+  - gke-cos-ingress-x.y
+  - gke-cos-reboot-x.y
+  - gke-gpu-device-plugin-x.y
+  - gke-gpu-device-plugin-p100-x.y
 * AWS
-  - kops-aws
+  - kops-aws-x.y
 * Node
-  - node-kubelet
+  - node-kubelet-x.y
 * [Kubeadm](https://github.com/kubernetes/kubeadm/blob/master/docs/managing-e2e-tests.md)
-  - kubeadm
-  - kubeadm-$(X-1)-on-$(X)
+  - kubeadm-gce
+  - kubeadm-gce-x.$(y-1)-on-x.y
 
 ## Testgrid Release Blocking Dashboards
```

Comment:
What about the upgrade jobs and the upgrade dashboards ? I understand they are much more thorny. do we want to tackle them separately? FWIW, @justinsb work on splitting the gce and gke upgrade jobs into (i) Serial & Disruptive and (ii) Slow (parallel) jobs have helped in stabilizing the latter.

Reply:
Practically speaking I think they should remain separate from master while we focus on them.  It may be that we decide to move upgrade jobs into blocking/informing down the line.

---

I don't want to bikeshed on it, but just to document since I went looking:

Previous releases use
- release-blocking
- release-all

And then the release-all dashboards fall out of maintenance and are almost always red (usually because of upgrade jobs)

What we current have with master is
- master-blocking
- master-upgrade
- master-upgrade-optional
- master-kubectl-skew
- misc

I haven't audited whether everything not in blocking ~= master-all, I would hope so but suspect not.  And then we have the fact that full scale scalability testing is never done on other release branches, only master.